src/instruction.rs

```diff
@@ -479,7 +479,7 @@ impl fmt::Display for Instruction {
             } => {
                 let mut parameter_str: String = parameters
                     .iter()
-                    .map(|p| p.to_string())
+                    .map(|p| format!("%{}", p.to_string()))
```

Comment:
Are both `format!` and `to_string()` necessary?

Reply:
good point ^